src/Glpi/UI/IllustrationManager.php

```diff
@@ -34,114 +34,55 @@
 
 namespace Glpi\UI;
 
-use DirectoryIterator;
+use Glpi\Application\View\TemplateRenderer;
 
 final class IllustrationManager
 {
+    public string $icons_definition_file;
+    public string $icons_sprites_path;
+
     public function __construct(
-        private string $illustration_dir = GLPI_ROOT . "/resources/illustration"
+        ?string $icons_definition_file = null,
+        ?string $icons_sprites_path = null,
     ) {
+        /** @var array $CFG_GLPI */
+        global $CFG_GLPI;
+
+        $this->icons_definition_file = $icons_definition_file ?? GLPI_ROOT
+            . '/public/lib/glpi-project/illustrations/icons.json'
+        ;
+        $this->icons_sprites_path = $icons_sprites_path ?? '/lib/glpi-project/illustrations/glpi-illustrations.svg'
+        ;
     }
 
     /**
-     * @param int $size Height and width (px)
+     * @param int|null $size Height and width (px). Will be set to 100% if null.
      */
-    public function render(string $filename, int $size = 100): string
+    public function renderIcon(string $icon_id, ?int $size = null): string
     {
-        if (!$this->isValidIllustrationName($filename)) {
-            return "";
-        }
-
-        $svg_content = $this->getSvgContent($filename);
-        $svg_content = $this->replaceColorsByVariables($svg_content);
-        $svg_content = $this->adjustSize($svg_content, $size);
-
-        return $svg_content;
+        $twig = TemplateRenderer::getInstance();
+        return $twig->render('components/illustration/icon.svg.twig', [
+            'file_path' => $this->icons_sprites_path,
+            'icon_id'   => $icon_id,
+            'size'      => $this->computeSize($size),
+        ]);
     }
 
     /** @return string[] */
-    public function getAllIllustrationsNames(): array
-    {
-        $illustrations = [];
-        $illustrations_files = new DirectoryIterator($this->getIllustrationDir());
-        foreach ($illustrations_files as $file) {
-            /** @var \SplFileInfo $file */
-            if ($file->isDir()) {
-                continue;
-            }
-
-            if ($file->getExtension() !== 'svg') {
-                continue;
-            }
-
-            $illustrations[] = $file->getFilename();
-        }
-
-        return $illustrations;
-    }
-
-    private function isValidIllustrationName(string $filename): bool
+    public function getAllIconsIds(): array
     {
-        $full_path = $this->getIllustrationDir() . "/$filename";
+        $json = file_get_contents($this->icons_definition_file);
+        $definition = json_decode($json, associative: true);
```

Comment:
When reading files, the code should be as defensive as possible.

```suggestion
        $json = file_get_contents($this->icons_definition_file);
        if ($json === false) {
            throw new \RuntimeException();
        }
        $definition = json_decode($json, associative: true, flags: JSON_THROW_ON_ERROR);
```